frontend/lib/src/components/core/Block/Block.tsx

```diff
@@ -182,6 +190,11 @@ const ChildRenderer = (props: BlockPropsWithWidth): ReactElement => {
 // Currently, only VerticalBlocks will ever contain leaf elements. But this is only enforced on the
 // Python side.
 const VerticalBlock = (props: BlockPropsWithoutWidth): ReactElement => {
+  const { setHideFullScreenButton } = React.useContext(LibContext)
+  if (props.hideFullScreenButton !== undefined) {
+    setHideFullScreenButton(props.hideFullScreenButton)
+  }
```

Comment:
Not sure I understand this part - why we are plumbing down `hideFullScreenButton` through VerticalBlock props and then setting it to the value we plumbed down here?

Reply:
That's a good point. I was under the assumption that someone who uses `@streamlit/lib` wanted to pass it down through the `VerticalBlock` but they can just pass it through the `LibContext.Provider`